jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridViewImpl.java

```diff
@@ -974,4 +977,148 @@ public void saveRefreshValue( int newValue ) {
     }
 
 
+    private boolean isColumnAdded( List<ColumnMeta<TaskSummary>> columnMetas,
+            String caption ) {
+        if ( caption != null ) {
+            for ( ColumnMeta<TaskSummary> colMet : columnMetas ) {
+                if ( caption.equals( colMet.getColumn().getDataStoreName() ) ) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public void addDomainSpecifColumns( ExtendedPagedTable<TaskSummary> extendedPagedTable,
+            Set<String> columns ) {
+
+        extendedPagedTable.storeColumnToPreferences();
+
+        HashMap modifiedCaptions = new HashMap<String, String>();
+        ArrayList<ColumnMeta> existingExtraColumns = new ArrayList<ColumnMeta>();
+        for ( ColumnMeta<TaskSummary> cm : extendedPagedTable.getColumnMetaList() ) {
+            if ( cm.isExtraColumn() ) {
+                existingExtraColumns.add( cm );
+            } else if ( columns.contains( cm.getCaption() ) ) {      //exist a column with the same caption
+                for ( String c : columns ) {
+                    if ( c.equals( cm.getCaption() ) ) {
+                        modifiedCaptions.put( c, "Var_" + c );
+                    }
+                }
+            }
+        }
+        for ( ColumnMeta colMet : existingExtraColumns ) {
+            if ( !columns.contains( colMet.getCaption() ) ) {
+                extendedPagedTable.removeColumnMeta( colMet );
+            } else {
+                columns.remove( colMet.getCaption() );
+            }
+        }
+
+        List<ColumnMeta<TaskSummary>> columnMetas = new ArrayList<ColumnMeta<TaskSummary>>();
+        String caption = "";
+        for ( String c : columns ) {
+            caption = c;
+            if ( modifiedCaptions.get( c ) != null ) {
+                caption = (String) modifiedCaptions.get( c );
+            }
+            Column genericColumn = initGenericColumn( c );
+            genericColumn.setSortable( false );
+
+            columnMetas.add( new ColumnMeta<TaskSummary>( genericColumn, caption, true, true ) );
+        }
+
+        extendedPagedTable.addColumns( columnMetas );
+
+    }
+
+
+    @Override
+    public FilterSettings getVariablesTableSettings( String taskName ) {
```

Comment:
Could we consider using a more object oriented style instead of using plain Strings? I guess something like JSONObject from GWT: http://www.gwtproject.org/javadoc/latest/com/google/gwt/json/client/JSONObject.html#toString%28%29